rustls-mbedcrypto-provider/src/kx.rs

```diff
@@ -28,110 +30,181 @@ use rustls::ffdhe_groups;
 use rustls::ffdhe_groups::FfdheGroup;
 use rustls::Error;
 use rustls::NamedGroup;
-/// A key-exchange group supported by *mbedtls*.
+/// A wrapper type representing implmentation of an EC key-exchange group
```

Comment:
implmentation -> implementation

Also this is not specific to EC, and this comment doesn't explain much.

Reply:
Huh, looking at the code, this is indeed EC, so this is a naming issue. So, this struct should be called `EcdhKxGroupWrapper` or so. And then, `DheKxGroupWrapper` should be `FfdheKxGroupWrapper`.

Could you try unify both implementations? So that `KxGroup` is able to represent both ECDH and FFDH? So it can be called `KxGroup`

Reply:
I will update them to use better names. 

Regarding the unification. I did not see the strong reason to unify them into one type especially why the algorithm and upstream trait implementation are different.

Note: name of `KxGroup` is first introduced when ffdhe is not added, so I think the problem here is we just need better naming for two types.